AirCasting/SDSync/MobileSessions/SDCardMobileSessionFinisher.swift

```diff
@@ -0,0 +1,27 @@
+// Created by Lunar on 4.06.25.
+//
+
+import Foundation
+import Resolver
+import CoreData
+
+
+protocol SessionFinisher {
+    func callAsFunction(uuid: SessionUUID) throws
+}
+
+class SDCardMobileSessionFinisher : SessionFinisher {
+    @Injected private var persistenceController: PersistenceController
+    private lazy var context: NSManagedObjectContext = persistenceController.editContext
+    
+    func callAsFunction(uuid: SessionUUID) throws {
```

Comment:
I'm honestly not a fan of this option in language. I know it's popular, so it's not even a change suggestion, just an academic discussion :D 
but having usage of 
`SessionFinisher.finishSession() 
instead of calling a function named (potentialy) differently in each file that it is being used seems more clear and with less potential to break during refactor or sth. 

But I know it's popular now, I just don't see the upside of that.